### PR TITLE
ci(action): GH action to validate PR title against conventional commit standard

### PR DESCRIPTION
**Pull request by sethkfman**
## **Description**

Added a GH action step to validate the PR title against the Conventional Commit standard 
https://www.conventionalcommits.org/en/v1.0.0/

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/19229?quickstart=1)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by owencraston — DISMISSED** [review]
I love this. Is there a way that we can link to the conventional commits doc in the case that it fails?

**Review by pedronfigueiredo — DISMISSED** [review]

File: .github/workflows/pr-title-linter.yml
Comment (PeterYinusa) [inline .github/workflows/pr-title-linter.yml:5]: Should this be `master` as we don't have a `main` branch on the extension?

Comment (sethkfman) [reply .github/workflows/pr-title-linter.yml]: Updated

**Review by danjm — CHANGES_REQUESTED** [review]
I want to propose the addition of some documentation along with this PR. I can write that documentation. I will do so within the next day or two and push a commit to this branch

**Comment by danjm** [comment]
> I will do so within the next day or two and push a commit to this branch

so I did not do that within a day or two

**Comment by HowardBraham** [comment]
I don't think this GitHub Action would run at the correct time.  It's set to run only on `master` and `develop`.  I believe it needs to run on any branch.

Also, let's update the library to 5.4.0

File: .github/workflows/pr-title-linter.yml
Comment (davidmurdoch) [inline .github/workflows/pr-title-linter.yml:1]: nit:

Most of out github actions sentences, like

Remove labels after issue (or PR) closed
Close stale issues and PRs
Trigger MetaMask Desktop CI
Update LavaMoat policies

So maybe we do that here, too?


```suggestion
name: "Validation PR Title adheres to Conventional Commits"
```

Also, the action's yml file name generally follow the `name` property. So it wouldn't hurt to rename it to resemble whatever action `"name"` you end up with.

**Review by davidmurdoch — DISMISSED** [review]
Note: I did _not_ test this on a separate repo to see if it works.

If it does in fact work, and it helps automate releases, I'm all for it! (we used a similar GH action for the Ganache repo and managed to automate about 95% of our release process, including writing release notes, example: https://github.com/trufflesuite/ganache/releases/tag/v7.6.0)

**Comment by HowardBraham** [comment]
@danjm @brad-decker Someone with admin access needs to add `amannn/action-semantic-pull-request@v5.4.0` to the allowed actions for this PR to work

**Review by HowardBraham — DISMISSED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by danjm — CHANGES_REQUESTED** [review]
Forgive me as I block 1 more time.

When introducing changes that cause PR builds to fail, I like to give our developers some notice.

I am going to put a message in the appropriate slack channels and say that we will merge this at the very end of this week, so that starting next week people have to format their PR titles this way.

So we can merge at end of day Friday.

File: .github/guidelines/LABELING_GUIDELINES.md
Comment (danjm) [inline .github/guidelines/LABELING_GUIDELINES.md:4]: This comment should probably go in a different file, as it is not about a PR label, but a PR title.

Perhaps we should create a new `PR_GUIDELINES.md`  file in the same directory, or add a file in the `docs/` directory

Comment (danjm) [reply .github/guidelines/LABELING_GUIDELINES.md:4]: Actually, adding it to the `CONTRIBUTING.md` file probably makes the most sense

File: .github/workflows/validate-conventional-commits.yml
Comment (danjm) [inline .github/workflows/validate-conventional-commits.yml:10]: Can we update this to use the same version that mobile is using `amannn/action-semantic-pull-request@c3cd5d1ea3580753008872425915e343e351ab54`

"As a general rule we don't use tags for actions, as a precaution against the action author's account being compromised. Tags can be moved, hashes cannot."

File: .github/guidelines/LABELING_GUIDELINES.md
Comment (HowardBraham) [reply .github/guidelines/LABELING_GUIDELINES.md]: Moved it to `CONTRIBUTING.md`, and also made several other changes to `CONTRIBUTING.md`:
- deleted `Add to the CHANGELOG` -- we're not doing this anymore
- `Close the issue` -- edited to match the new PR template
- `Get reviewed by a core contributor` -- edited to `**Get reviewed by MetaMask Internal Developers**: All PRs require 2 approvals from MetaMask Internal Developers before merging.`
- `LABELING_GUIDELINES` -- fixed the broken hyperlink

File: .github/workflows/validate-conventional-commits.yml
Comment (HowardBraham) [reply .github/workflows/validate-conventional-commits.yml]: Done

**Review by HowardBraham — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Comment by HowardBraham** [comment]
@davidmurdoch I think you're right, I think it's not working 100% of the time.  Some of this may be PRs that were opened before this merged.

**Pull request merged**
